src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -959,8 +964,14 @@ internal static String Format(DateTime dateTime, String format, DateTimeFormatIn
             }
 
             if (format.Length == 1) {
-                if (format[0] == 'o' || format[0] == 'O') {
+                switch (format[0])
+                {
+                    case 'O':
+                    case 'o':
                         return FastFormatRoundtrip(dateTime, offset);
+                    case 'R':
+                    case 'r':
```

Comment:
I am seeing in the ExpandPredefinedFormat we have the following code, but it looks you don't handle that 

                case 'r':
                case 'R':       // RFC 1123 Standard                    
                    if (offset != NullOffset) {
                        // Convert to UTC invariants mean this will be in range
                        dateTime = dateTime - offset;
                    }
                    else if (dateTime.Kind == DateTimeKind.Local) {
                        InvalidFormatForLocal(format, dateTime);
                    }
                    dtfi = DateTimeFormatInfo.InvariantInfo;
                    break;